apps/fxc-front/src/app/logic/score/scorer.ts

```diff
@@ -1,19 +1,101 @@
-import { CircuitType } from '@flyxc/optimizer/lib/api';
+import type { LatLonAltTime, ScoringResult } from '@flyxc/optimizer/lib/optimizer';
 
-export class Score {
-  distanceM: number;
-  indexes: number[];
-  multiplier: number;
-  circuit: CircuitType;
-  closingRadiusKm: number;
-  points: number;
+import { type Response as WorkerResponse } from '../../workers/optimizer';
+import ScoringWorker from '../../workers/optimizer?worker';
+import { getScoringRuleName, type LeagueCode } from './league/leagues';
 
-  constructor(score: Partial<Score>) {
-    this.distanceM = score.distanceM ?? 0;
-    this.indexes = score.indexes ?? [];
-    this.multiplier = score.multiplier ?? 1;
-    this.circuit = score.circuit ?? CircuitType.OpenDistance;
-    this.closingRadiusKm = score.closingRadiusKm ?? 0;
-    this.points = score.points ?? 0;
+export type ScoringResultHandler = (result: ScoringResult) => void;
```

Comment:
Note: add some point we will change the type of the param, i.e.:

```
{
  type: 'ok' | 'cancelled' | 'timeout' | 'error';
  result?: ScoringResult;
}
```

not needed now